python/Ganga/Utility/Config/Config.py

```diff
@@ -157,18 +157,19 @@ def f(*args, **kwds):
 def _migrate_name(name):
     import Ganga.Utility.strings as strings
 
-    if (name not in translated_names.keys()) and (not strings.is_identifier(name)):
+    if (name not in translated_names) and (not strings.is_identifier(name)):
         name2 = strings.drop_spaces(name)
-        name2 = name2.replace(':', '_')
+        name3 = name2.replace(':', '_')
 
-        if not strings.is_identifier(name2):
-            raise ValueError(
-                'config name %s is not a valid python identifier' % name)
+        if not strings.is_identifier(name3):
+            raise ValueError('config name %s is not a valid python identifier' % name)
         else:
             logger = getLogger()
-            logger.warning('obsolete config name found: replaced "%s" -> "%s"' % (name, name2))
+            logger.warning('obsolete config name found: replaced "%s" -> "%s"' % (name, name3))
             logger.warning('config names must be python identifiers, please correct your usage in the future ')
-            translated_names[name] = name2
+            translated_names[name] = name3
+    elif name in translated_names:
+        pass
```

Comment:
True, but that is a *very* micro optimisation being performed by a strangely written bit of code with no comment to explain it. Code should express intent first and only worry about optimisations like this is they are really going to be a problem. Over the course of a whole run of the program I guess this function might, at most, be run a few hundred times (~360 by my count) which will add up to milliseconds in efficiency gains at most. It is simply not worth it.

What is worth it is looking at the slightly bigger picture and realising that this function `_migrate_name()` is here to convert very old config section names from code going back to older than the repo history goes (>6 years). Not once does it actually seem to ever change any names. If you want to increase efficiency then simply removing this function entirely will do a much better job.